plugin/discovery/error.go

```diff
@@ -33,3 +44,71 @@ const ErrorNoVersionCompatibleWithPlatform = Error("no available version is comp
 func (err Error) Error() string {
 	return string(err)
 }
+
+// VersionIncompatibleError is a wrapper type for the error const ErrorVersionIncompatible
+// that includes a DetailedErrorMsg method to display a custom detailed message pertaining to
+// the incompatible version
+type VersionIncompatibleError Error
+
+func (err VersionIncompatibleError) Error() string {
+	return ErrorVersionIncompatible.Error()
```

Comment:
I found myself staring at this for quite some time before I realized that `ErrorVersionCompatible` and `VersionIncompatibleError` are not the same thing here. :confounded: 

I'm not sure what other name to suggest here, though... it's tough that the prior code was assuming individual error objects with constant strings, and I see why this was important.

Perhaps this is a reasonable situation to use [`errwrap`](https://godoc.org/github.com/hashicorp/errwrap) to wrap the `ErrorVersionIncompatible` object without needing to add a new type, etc?

```go
    return errwrap.Wrapf(ErrorVersionIncompatible, fmt.Sprintf(msg, provider, selected.String, etcetc))
```

The caller could then use `errwrap.Contains` to test the wrapping, maybe? It feels a bit clumsy but maybe a little easier to follow? I'm not sure.

Reply:
Totally understandable on the readability of the names/code. I took another jab at refactoring it and found a decent approach with using `errorwrap` that you had suggested.

Also updated the error printing format to be more similar to the UI errors for `init` (first line is bold and red w/o "Error" prefix)